hoomd/data/typeparam.py

```diff
@@ -197,6 +197,11 @@ def _attach(self, cpp_obj, sim):
             self.param_dict)
         return self
 
+    def _attach_with_mesh(self, cpp_obj, mesh):
+        self.param_dict = AttachedTypeParameterDict(
+            cpp_obj, self.name, getattr(mesh, self.type_kind), self.param_dict)
+        return self
+
```

Comment:
I would really like to find a way to make this unnecessary.

Reply:
One way could be to abstract out the way to query for types to the constructor. There should be a way to do this by either extending the abstraction or fitting this into the existing one.